independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/QuarkusClassLoader.java

```diff
@@ -233,6 +233,8 @@ public void reset(Map<String, byte[]> generatedResources, Map<String, byte[]> tr
             throw new IllegalStateException("Classloader is not resettable");
         }
         synchronized (this) {
+            // we don't want the previous MemoryClassPathElement to leak as a key of protectionDomains
+            protectionDomains.remove(this.transformedClasses);
```

Comment:
This doesn't look right, transformedClasses is a `Map<String, byte[]>` and protectionDomains is a `ConcurrentMap<ClassPathElement, ProtectionDomain>`

Reply:
Perhaps the parameter name can be renamed to something less confusing, but that's not priority